agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/presto/PluginClassLoaderTracer.java

```diff
@@ -46,15 +50,35 @@ protected Implementation newImplementation() {
 
         @Advice.OnMethodExit
         public static void addGarmadonAgentPackage(@Advice.This Object o) throws Exception {
-            Class pluginClassLoaderZz = o.getClass().getClassLoader().loadClass("com.facebook.presto.server.PluginClassLoader");
-            Field spiClassLoaderField = pluginClassLoaderZz.getDeclaredField("spiPackages");
-            spiClassLoaderField.setAccessible(true);
 
-            ImmutableList<String> spiPackages = (ImmutableList<String>) spiClassLoaderField.get(o);
-            List<String> tmpspiPackages = new ArrayList<>(spiPackages);
-            tmpspiPackages.add("com.criteo.hadoop.garmadon.agent.");
+            /*
+             * If we fail to add the garmadon agent package to at least one PluginClassLoader
+             * Then we re-throw the exception raised by the last attempt that got catch while trying
+             */
+            boolean success = false;
+            Exception exception = null;
+
+            for (String pluginClassLoaderClassName : Arrays.asList(PRESTOSQL_PLUGINCLASSLOADER, PRESTODB_PLUGINCLASSLOADER)) {
+                try {
+                    Class pluginClassLoaderZz = o.getClass().getClassLoader().loadClass(pluginClassLoaderClassName);
+                    Field spiClassLoaderField = pluginClassLoaderZz.getDeclaredField("spiPackages");
+                    spiClassLoaderField.setAccessible(true);
+
+                    ImmutableList<String> spiPackages = (ImmutableList<String>) spiClassLoaderField.get(o);
+                    List<String> tmpspiPackages = new ArrayList<>(spiPackages);
+                    tmpspiPackages.add("com.criteo.hadoop.garmadon.agent.");
+
+                    spiClassLoaderField.set(o, ImmutableList.copyOf(tmpspiPackages));
+
+                    success = true;
+                } catch (Exception e) {
+                    exception = e;
+                }
+            }
 
-            spiClassLoaderField.set(o, ImmutableList.copyOf(tmpspiPackages));
+            if (!success) {
+                throw exception;
+            }
```

Comment:
What about just using o.getClass().getName() instead of pluginClassLoaderClassName to get the appropriate plugin name

`Class pluginClassLoaderZz = o.getClass().getClassLoader().loadClass(o.getClass().getName());
            Field spiClassLoaderField = pluginClassLoaderZz.getDeclaredField("spiPackages");
            spiClassLoaderField.setAccessible(true);

            ImmutableList<String> spiPackages = (ImmutableList<String>) spiClassLoaderField.get(o);
            List<String> tmpspiPackages = new ArrayList<>(spiPackages);
            tmpspiPackages.add("com.criteo.hadoop.garmadon.agent.");

            spiClassLoaderField.set(o, ImmutableList.copyOf(tmpspiPackages));